### PR TITLE
PixelPaint: Add an option for making a Gradient with a secondary color

### PR DESCRIPTION
**Pull request by krkk**
https://user-images.githubusercontent.com/16520278/218553187-9b2c8671-7950-4217-9da6-25f70635c055.mp4

### CODE REVIEW

**Comment by Torstennator** [comment]
Hey @krkk,
i like option to have the second color available for the gradient :)
Let my try to explain why I was 'holding' the transparent color at the side where the cursor is when you initially draw the gradient. 
My intentation was that drawing the colored side of the gradient at the opposite side where the cursor is moved is that you always see what you are going to cover with it while defining the gradient.
That way you could easily initially move the gradient to something on the image that you would not like to cover...

Basically i was just thinking from the other direction... i'm drawing the gradient until the point where the cursor is in the same direction as i was moving the cursor. And not that i'm pushing the gradient away with the cursor.

**Comment by AtkinsSJ** [comment]
Code looks fine to me. However, since we're already talking about how the cursor works:
I would expect clicking and then dragging, to set the main colour's position where the click starts, and then set the second colour's position where you release it. I checked and it's how GIMP works:
Peek 2023-02-15 14-21 gradients

That just seems the most intuitive to me, rather than the click being at the centre of the gradient.

**Comment by Torstennator** [comment]
I'm also fine with the changes - no reason to not merge  it ;-)

When i was building the gradient tool i was aiming for the functionality that CaptureOne has when creating alpha masks. So it really depends on the tools you are used to ;-)

PixelPaint_Gradient
